test/e2e/specs/02_admin.js

```diff
@@ -1,56 +1,43 @@
 module.exports = {
   '@tags': ['admin', 'login'],
-   beforeEach: (client) => {
 
+   before: (client) => {
     client.resizeWindow(1024, 800);
   },
+
+  afterEach: (client, done) => {
+    if (client.currentTest.results.failed) {
+      throw new Error('Test Case failed, skipping all the rest');
+    }
+    done();
+  },
+
+  after: (client) => {
+    client.end();
+  },
+
   'Admin logs in': (client) => {
     const adminPage = client.page.admin();
     const {testData: {admin}} = client.globals;
 
-    adminPage
-      .navigate()
-      .waitForElementVisible('@loginLayout')
-      .waitForElementVisible('@signInForm')
-      .setValue('@emailInput', admin.email)
-      .setValue('@passwordInput', admin.password)
-      .waitForElementVisible('@signInButton')
-      .click('@signInButton');
-
-    adminPage
-      .waitForElementVisible('@moderationContainer');
+    adminPage.navigate();
+    adminPage.expect.section('@login').to.be.visible;
+    adminPage.section.login.login(admin);
```

Comment:
Why did you move the login command inside the login section? It should be a general command, IMO. `admin.login(userData)` => To login inside the admin-side no matter where you are.

Reply:
- I'm splitting the pages into logical parts and the top level should only contain elements that are almost always available. The login section is only available when you are not logged in.
- I'm avoiding using `navigate` as much as possible to simulate a user clicking through the page, also decreases testing time and increases stability of the tests.
- The commands I added are stateless building blocks of user interaction that you can do at a certain section. Think of it as more lower level commands.
- We can provide a top level command called `navigateAndLogin` that calls the lower level commands for you.

Reply:
I'll add a `navigateAndLogin` method.